### PR TITLE
docs: edit dependency-injection.md

### PR DESCRIPTION
**Pull request by kapunahelewong**
Moves Injector hierarchy and service instances section into providers doc. And moves Dependency injection tokens
section to dependency-injection-providers.md. Removes material that is duplicated in other documents. Focuses this document to more clearly be a task-based doc. Brings in line with styles.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/dependency-injection.md [hunk 1]
```diff
@@ -1,217 +1,81 @@
...
-The `@Injectable()` is an essential ingredient in every Angular service definition. The rest of the class has been written to expose a `getHeroes` method that returns the same mock data as before. (A real app would probably get its data asynchronously from a remote server, but we'll ignore that to focus on the mechanics of injecting the service.)
+The `@Injectable()` decorator specifies that Angular can use this class in the DI system.
+The metadata, `providedIn: 'root'`, means that the `HeroService` is visible throughout the application.
+
+Next, to get the hero mock data, add a `getHeroes()` method that returns the heroes from `mock.heroes.ts`.
 
 <code-example path="dependency-injection/src/app/heroes/hero.service.3.ts" header="src/app/heroes/hero.service.ts">
 </code-example>
 
+For clarity and maintainability, it is recommended that you define components and services in separate files.
```
Comment (jelbourn) [inline aio/content/guide/dependency-injection.md:36]: Pass voice?

**Review by jelbourn — APPROVED** [review]
Overall LGTM, couple of minor comments

File: aio/content/guide/dependency-injection-providers.md [hunk 2]
```diff
@@ -24,6 +24,29 @@ For instance:
...
+
+<code-example path="dependency-injection/src/app/heroes/hero-list.component.ts" region="ctor-signature" header="src/app/heroes/hero-list.component.ts">
+</code-example>
+
+Though classes provide many dependency values, the expanded `provide` object lets you associate different kinds of providers with a DI token.
+
+
 {@a provide}
 
 ## The `Provider` object literal
```
Comment (aikidave) [inline aio/content/guide/dependency-injection-providers.md:52]:
```suggestion
## Defining providers
```

File: aio/content/guide/dependency-injection-providers.md [hunk 3]
```diff
@@ -40,15 +63,14 @@ The following code snippets shows how a class that is given as the `providers` v
...
 
 * The second property is a provider definition object, which tells the injector how to create the dependency value.
 The provider-definition key can be `useClass`, as in the example.
 It can also be `useExisting`, `useValue`, or `useFactory`.
 Each of these keys provides a different type of dependency, as discussed below.
 
-
 {@a class-provider}
 
 ## Alternative class providers
```
Comment (aikidave) [inline aio/content/guide/dependency-injection-providers.md:76]:
```suggestion
## Specifying an alternative class provider
```

**Review by aikithoughts — COMMENTED** [review]
Can we make sure that if we are including content in the Main Concepts, it is task-based/task-focused? A lot of this content is very conceptual. I find myself asking: "What am I supposed to do? How does this help me?"

Thoughts?

**Comment by kapunahelewong** [comment]
> Can we make sure that if we are including content in the Main Concepts, it is task-based/task-focused? A lot of this content is very conceptual. I find myself asking: "What am I supposed to do? How does this help me?"
> 
> Thoughts?

Sure @aikidave, the edits to `dependency-injection-providers.md` in this PR (#39255) are only to move the sections on tokens and the object literal that didn't fit in `dependency-injection.md`, rather than editing them. The edits to `dependency-injection-providers.md` are in https://github.com/angular/angular/pull/39403. I've also made your updates in #39403, but most of the original copy is no longer in #39403 as i have edited them there. Does that make sense?

**Review by aikithoughts — APPROVED** [review]
LGTM

Reviewed-for: global-docs-approvers

**Pull request closed without merging**